File: app/src/main/java/com/thelittlefireman/appkillermanager_exemple/MainActivity.java
package com.thelittlefireman.appkillermanager_exemple;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import androidx.appcompat.widget.AppCompatCheckBox;
import android.view.View;
import android.widget.Button;

import com.thelittlefireman.appkillermanager.managers.KillerManager;
import com.thelittlefireman.appkillermanager.ui.DialogKillerManagerBuilder;
import com.thelittlefireman.appkillermanager.models.KillerManagerActionType;
import com.thelittlefireman.appkillermanager.utils.LogUtils;

import butterknife.BindView;
import butterknife.ButterKnife;

public class MainActivity extends Activity {
    @BindView(R.id.powerSavingManagerButton)
    Button powerSavingManagerButton;
    @BindView(R.id.autoStartManagerButton)
    Button autoStartManagerButton;
    @BindView(R.id.notificationManagerButton)
    Button notificationManagerButton;

    @BindView(R.id.idByDialog)
    AppCompatCheckBox mAppCompatCheckBoxByDialog;

    private KillerManagerActionType mAction;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        KillerManager.init(this);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);
        KillerManager.init(this);
        LogUtils.registerLogCustomListener(new LogUtils.LogCustomListener() {
            @Override
            public void i(String tag, String message) {
                // Custom Log
            }

            @Override
            public void e(String tag, String message) {
                // Custom Log
            }
        });
        powerSavingManagerButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (mAppCompatCheckBoxByDialog.isChecked()) {
                    mAction = KillerManagerActionType.ACTION_POWERSAVING;
                    startDialog(mAction);
                } else {
                    KillerManager.doAction(MainActivity.this, mAction);
                }
            }
        });
        autoStartManagerButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (mAppCompatCheckBoxByDialog.isChecked()) {
                    mAction = KillerManagerActionType.ACTION_AUTOSTART;
                    startDialog(mAction);
                } else {
                    KillerManager.doAction(MainActivity.this, mAction);
                }
            }
        });
        notificationManagerButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (mAppCompatCheckBoxByDialog.isChecked()) {
                    mAction = KillerManagerActionType.ACTION_NOTIFICATIONS;
                    startDialog(mAction);
                } else {
                    KillerManager.doAction(MainActivity.this, mAction);
                }
            }
        });
    }

    public void startDialog(KillerManagerActionType action) {

        new DialogKillerManagerBuilder().setActivity(this).setAction(action).show();

    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        KillerManager.onActivityResult(MainActivity.this, mAction,requestCode);
    }
}

File: app/src/main/java/com/thelittlefireman/appkillermanager_exemple/MainActivity.kt
package com.thelittlefireman.appkillermanager_exemple

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.thelittlefireman.appkillermanager.devices.DeviceBase
import com.thelittlefireman.appkillermanager.managers.KillerManager
import com.thelittlefireman.appkillermanager.models.KillerManagerActionType
import com.thelittlefireman.appkillermanager.ui.DialogKillerManager
import com.thelittlefireman.appkillermanager.utils.LogUtils
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    private lateinit var actionType: KillerManagerActionType

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        LogUtils.registerLogCustomListener(object : LogUtils.LogCustomListener {
            override fun i(tag: String, message: String) {
                // Custom Log
            }

            override fun w(tag: String, message: String) {
                // Custom log
            }

            override fun e(tag: String, message: String?) {
                // Custom Log
            }
        })

        powerSavingManagerButton.setOnClickListener {
            val actionType = KillerManagerActionType.ActionPowerSaving
                .also { actionType = it }

            if (idByDialog.isChecked)
                startDialog(actionType)
            else
                KillerManager.doAction(this, actionType)
        }
        autoStartManagerButton.setOnClickListener {
            val actionType = KillerManagerActionType.ActionAutoStart
                .also { actionType = it }

            if (idByDialog.isChecked)
                startDialog(actionType)
            else
                KillerManager.doAction(this, actionType)
        }
        notificationManagerButton.setOnClickListener {
            val actionType = KillerManagerActionType.ActionNotifications
                .also { actionType = it }

            if (idByDialog.isChecked)
                startDialog(actionType)
            else
                KillerManager.doAction(this, actionType)
        }

        val device: DeviceBase? = KillerManager.device

        if (device != null && device.isThatRom) {
            if (!device.isActionPowerSavingAvailable(this))
                powerSavingManagerButton.optimisationNotAvailable()

            if (!device.isActionAutoStartAvailable(packageManager))
                autoStartManagerButton.optimisationNotAvailable()

            if (!device.isActionNotificationAvailable())
                notificationManagerButton.optimisationNotAvailable()
        } else {
            powerSavingManagerButton.optimisationNotAvailable()
            autoStartManagerButton.optimisationNotAvailable()
            notificationManagerButton.optimisationNotAvailable()
        }
    }

    private fun startDialog(action: KillerManagerActionType) =
        DialogKillerManager(
            activity = this,
            killerManagerActionTypeList = listOf(action),
            titleMessage = "startDialog"
        )
            .show()

    private fun Button.optimisationNotAvailable() {
        isEnabled = false
        setText(R.string.optimization_not_available)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        KillerManager.onActivityResult(this, actionType, requestCode)
    }
}
